tstate/tstate_test.go

```diff
@@ -323,3 +325,46 @@ func TestWriteChanges(t *testing.T) {
 		require.ErrorIs(err, database.ErrNotFound, "Value not removed from db.")
 	}
 }
+
+func BenchmarkFetchAndSetScope(b *testing.B) {
+	for _, size := range []int{10, 100, 1000} {
+		b.Run(fmt.Sprintf("fetch_and_set_scope_%d_keys", size), func(b *testing.B) {
+			benchmarkFetchAndSetScope(b, size)
+		})
+	}
+}
+
+func benchmarkFetchAndSetScope(b *testing.B, size int) {
+	require := require.New(b)
+	ts := New(10)
+	db := NewTestDB()
+	ctx := context.TODO()
+
+	keys := [][]byte{}
+	vals := [][]byte{}
+
+	// each k/v is unique to simulate worst case
+	for range "0..size" {
```

Comment:
I don't think this does what you think it does :)

Reply:
haha